usb.c

```diff
@@ -1200,13 +1200,70 @@ static int usb_fill_context_info(struct iio_context_info *info,
 	return 0;
 }
 
-int usb_context_scan(struct iio_scan_result *scan_result)
+#define SCAN_WILDCARD "*"
+#define SCAN_DELIMITER ','
+
+static int parse_value(const char *str)
+{
+	uint32_t val;
+	char *ptr;
+
+	/* if it is null or '*' accept any */
+	if (str == NULL || !strncmp(str, SCAN_WILDCARD, sizeof(SCAN_WILDCARD)))
+		return 0;
+	/* if it's empty, that's not valid */
+	if (*str == '\0')
+		return -1;
+
+	val = strtoul(str, &ptr, 16);
+
+	if (ptr == str)
+		return -1;
+
+	/* should be either ":" or end of string */
+	if (*ptr != SCAN_DELIMITER && *ptr != '\0')
+		return -1;
+
+	if (val > 0xFFFF)
+		return -1;
+
+	return val;
+}
+
+int usb_context_scan(struct iio_scan_result *scan_result,
+		char *pid_vid)
 {
 	struct iio_context_info *info;
 	libusb_device **device_list;
 	libusb_context *ctx;
 	unsigned int i;
 	int ret;
+	uint16_t scan_usb_vid = 0, scan_usb_pid = 0;
+	char *ptr;
+
+	/* string must be either: "usb" or "usb=vid," or "usb=vid,*"
+	 * or "usb=vid,pid" format. Both ID's are given in hexadecimal.
+	 * No leading 0x is required
```

Comment:
You need to update the comment here, as we now use a colon as the delimiter for vid:pid.